### DISCUSSION TITLE
Using "required_if" validation rule with an enum value

### DISCUSSION THREAD

**Discussion by Bosphoramus** [laravel/framework#48500] [Q&A]
Is there a more elegant way to do this?

```php
'iban' => ['required_if:payment_method:'.PaymentMethod::DebitReceipt->value]
```

I want to use the required_if validation rule to check if another field in the request is equal to a value contained in an enum.

My code right now:

```php
'payment_method' => ['required', new Enum(PaymentMethod::class)],
'iban' => ['required_if:payment_method:'.PaymentMethod::DebitReceipt->value],
```

I have categorised this as a Q&A because I do not know if there is already a better (more Laravel-like) way to do this, if not it could be categorised as a feature idea.

Thank you!

**Comment by roelvanhintum** [comment]
For future reference, `required_if` doesn't support enums. This would work as follows (in my case in a GraphQL mutation):
```php
'rules' => fn (array $args) => [
    Rule::requiredIf(function () use ($args) {
        return $args['payment_method'] === PaymentMethod::DebitReceipt;
    }),
],
```
